tests/ops/api/v1/endpoints/test_connection_config_endpoints.py

```diff
@@ -107,6 +100,106 @@ def test_patch_http_connection(
         http_config = ConnectionConfig.get_by(db, field="key", value="webhook_key")
         http_config.delete(db)
 
+    def test_patch_connection_saas_with_secrets_exists(
+        self, url, api_client, generate_auth_header, db, mailchimp_config
+    ):
+        payload = [
+            {
+                "secrets": {
+                    "domain": "test_mailchimp_domain",
+                    "username": "test_mailchimp_username",
+                    "api_key": "test_mailchimp_api_key",
+                },
+                "name": "My Mailchimp Test",
+                "description": "Mailchimp ConnectionConfig description",
+                "key": f"mailchimp{uuid4()}",
+                "connection_type": "saas",
+                "access": "read",
+            },
+        ]
+
+        ConnectionConfig.create(
+            db=db,
+            data={
+                "key": payload[0]["key"],
+                "name": payload[0]["name"],
+                "connection_type": ConnectionType.saas,
+                "access": AccessLevel.write,
+                "secrets": {"domain": "test", "username": "test", "api_key": "test"},
+                "saas_config": mailchimp_config,
+            },
+        )
+
+        auth_header = generate_auth_header(scopes=[CONNECTION_CREATE_OR_UPDATE])
+        response = api_client.patch(url, headers=auth_header, json=payload)
+        assert 200 == response.status_code
+        body = json.loads(response.text)
+        assert body["succeeded"][0]["connection_type"] == payload[0]["connection_type"]
+
+    def test_patch_connection_saas_with_secrets_new(
+        self, url, api_client, generate_auth_header
+    ):
+        payload = [
+            {
+                "secrets": {
+                    "domain": "test_mailchimp_domain",
+                    "username": "test_mailchimp_username",
+                    "api_key": "test_mailchimp_api_key",
+                },
+                "name": "My Mailchimp Test",
+                "description": "Mailchimp ConnectionConfig description",
+                "key": "mailchimp_1",
+                "connection_type": "saas",
+                "saas_connection_type": "mailchimp",
+                "access": "read",
+            },
+        ]
+
+        auth_header = generate_auth_header(scopes=[CONNECTION_CREATE_OR_UPDATE])
+        response = api_client.patch(url, headers=auth_header, json=payload)
+        assert 200 == response.status_code
+        body = json.loads(response.text)
+        assert body["succeeded"][0]["connection_type"] == payload[0]["connection_type"]
```

Comment:
could we go a bit further and assert here that the connection config is added to the db with the data we expect, including the `.secrets`? 

may be good to do some basic verification to make sure the `saas_config` is populated from the template on the create here too.